### ISSUE TITLE
Feature request - Object tree view (Database navigator) preloading on connection start

### ISSUE THREAD

**Issue by yevon** [dbeaver/dbeaver#2923]
It would be nice having an option to preload in background object lists in Database navigator on connection start. This option should be at connection level and configurable by object type ("Table", "View", "Sequence"), or enabled by default if the number of objects/type is relatively low (less than 200?).

This way initial responsivness would be even better

**Comment by serge-rider** [comment]
This makes sense. We could add extra config option for that + some object types configuration (not necessarily in the first version).

By default this will be turned off because DBeaver often used with remote databases over VPNs/tunnels where each server roundtrip is very expensive.

**Comment by yevon** [comment]
I agree with that, it should be disabled by default, because in some enviroments with very large databases or slow connections might be an issue. In many other situations would be a nice feature.Good!

**Comment by DavideArgellati** [comment]
definitely! having configurable to open the tables would be amazing, in this regards competitors handle it much better
